toolkit/components/uniffi-bindgen-gecko-js/src/render/cpp.rs

```diff
@@ -2,44 +2,72 @@
 License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use super::shared::*;
+use crate::{FunctionIds, ObjectIds};
 use askama::Template;
 use extend::ext;
-use heck::{ToLowerCamelCase, ToSnakeCase, ToUpperCamelCase};
-use uniffi_bindgen::interface::{
-    ComponentInterface, FFIArgument, FFIFunction, FFIType , Object,
-};
+use heck::ToUpperCamelCase;
+use std::iter;
+use uniffi_bindgen::interface::{ComponentInterface, FFIArgument, FFIFunction, FFIType, Object};
 
 #[derive(Template)]
-#[template(path = "Scaffolding.cpp", escape = "none")]
-pub struct CPPScaffoldingTemplate {
-    pub ci: ComponentInterface,
-}
-
-#[derive(Template)]
-#[template(path = "Scaffolding.h", escape = "none")]
-pub struct CPPHeaderScaffoldingTemplate {
-    pub ci: ComponentInterface,
+#[template(path = "UniFFIScaffolding.cpp", escape = "none")]
+pub struct CPPScaffoldingTemplate<'a> {
+    // Namespace to put the functions in (under mozilla::dom).  This is related to how we handle
+    // scaffolding for the test fixtures.  For each function defined UniFFI.webidl we:
+    //   - Generate a function in `mozilla::dom::uniffi` to handle it using the real UDL files
```

Comment:
I'm sure the DOM team will have opinions here, but `mozilla::dom` doesn't seem quite right here - taking the `dom` out might also make them feel less like they are being asked to take ownership :)